### PR TITLE
wxpython 4.1.1

### PR DESCRIPTION
**Pull request by chenrui333**
Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by chenrui333** [comment]
```
Error: An exception occurred within a child process:
  Utils::Inreplace::Error: inreplace failed
buildtools/build_wxwidgets.py:
  expected replacement of "--with-macosx-version-min=10.9" with "--with-macosx-version-min=11"
```

File: Formula/wxpython.rb [hunk 1]
```diff
@@ -34,24 +33,34 @@ class Wxpython < Formula
...
+
   def install
     # Fix build of included wxwidgets:
     # https://github.com/wxWidgets/Phoenix/issues/1247
     # https://github.com/Homebrew/homebrew-core/pull/58988
     inreplace "buildtools/build_wxwidgets.py" do |s|
       s.gsub! "#wxpy_configure_opts.append(\"--enable-monolithic\")",
               "wxpy_configure_opts.append(\"--disable-precomp-headers\")"
-      s.gsub! "--with-macosx-version-min=10.9",
+      s.gsub! "--with-macosx-version-min=10.10",
```
Comment (fxcoudert) [inline Formula/wxpython.rb:63]: The `--with-macosx-version-min` fix has been committed upstream (but not the other one)

File: Formula/wxpython.rb [hunk 2]
```diff
@@ -34,24 +33,34 @@ class Wxpython < Formula
...
+    url "https://files.pythonhosted.org/packages/2b/06/93bf1626ef36815010e971a5ce90f49919d84ab5d2fa310329f843a74bc1/Pillow-8.0.1.tar.gz"
+    sha256 "11c5c6e9b02c9dac08af04f093eb5a2f84857df70a7d4a6a6ad461aca803fb9e"
   end
 
   resource "six" do
     url "https://files.pythonhosted.org/packages/6b/34/415834bfdafca3c5f451532e8a8d9ba89a21c9743a0c59fbd0205c7f9426/six-1.15.0.tar.gz"
     sha256 "30639c035cdb23534cd4aa2dd52c3bf48f06e5f4a941509c8bafd8ce11080259"
   end
 
+  resource "wxPython" do
```
Comment (fxcoudert) [inline Formula/wxpython.rb:51]: I don't think that's right, downloading the source twice

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/wxpython.rb (hunk 2 above)
Comment (chenrui333) [reply Formula/wxpython.rb:51]: yeah, you are right, removing this.

File: Formula/wxpython.rb (hunk 1 above)
Comment (chenrui333) [reply Formula/wxpython.rb:63]: Got it.

**Comment by fxcoudert** [comment]
Why do we install numpy as a resource if we depend on the numpy formula?

**Comment by chenrui333** [comment]
@fxcoudert that is actually generated from `brew bump-formula-pr`, I'd prefer to use the generated resource, as it is derived dependency (might be different version from numpy formula).

**Comment by fxcoudert** [comment]
> that is actually generated from brew bump-formula-pr

But in this case it is the wrong thing to do

> I'd prefer to use the generated resource

There does not seem to be a conflict, so we should avoid multiplying downloads and installs of the same thing for no reason. Moreover, numpy needs some specific setup to have reasonable performance, so we should definitely not include it simply as a resource.

**Comment by Rylan12** [comment]
For cases like these, we can now add exceptions for `update-python-resources` in [`pypi_formula_mappings.json`](https://github.com/Homebrew/homebrew-core/blob/master/pypi_formula_mappings.json).

Adding the following looks like it will solve this:

```json
  "wxpython": {
    "exclude_packages": ["numpy"]
  }
```

**Review by fxcoudert — APPROVED** [review]

**Pull request closed without merging**